sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -937,6 +937,14 @@ object SQLConf {
       .timeConf(TimeUnit.SECONDS)
       .createWithDefault(0L)
 
+  val THRIFTSERVER_FORCE_CANCEL =
+    buildConf("spark.sql.thriftServer.interruptOnCancel")
+      .doc("When true, all running tasks will be interrupted if one cancels a query. " +
+        "When false, all running tasks will be remained until finished.")
```

Comment:
Shall we guide users to enable this configuration when `spark.sql.thriftServer.queryTimeout` is set?

Reply:
+1

Reply:
+1

Reply:
yeah It sounds so warm !